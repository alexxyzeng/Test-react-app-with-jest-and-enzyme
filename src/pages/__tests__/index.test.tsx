import 'jest';
import Index from '..';
import React, { useState } from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer, { ReactTestInstance, ReactTestRenderer } from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe('Page: index', () => {
  it('Render correctly', () => {
    const wrapper: ReactTestRenderer = renderer.create(<Index />);
    expect(wrapper.root.children.length).toBe(1);
    const outerLayer = wrapper.root.children[0] as ReactTestInstance;
    expect(outerLayer.type).toBe('div');
    expect(outerLayer.children.length).toBe(2);
  });
});

describe('Page: index', () => {
  it('Render correctly', () => {
    const wrapper = mount(<Index />);
    expect(wrapper.children()).toHaveLength(1);
    const outerLayer = wrapper.childAt(0);
    expect(outerLayer.type()).toBe('div');
    expect(outerLayer.children()).toHaveLength(2);
  });
  it('case of use state', () => {
    const wrapper = shallow(<Index />);
    expect(wrapper.find('#intro').text()).toBe('Welcome to Umi');
    wrapper.find('button').simulate('click');

    expect(wrapper.find('#intro').text()).toBe('Welcome to Jest and Enzyme');
  });
});
