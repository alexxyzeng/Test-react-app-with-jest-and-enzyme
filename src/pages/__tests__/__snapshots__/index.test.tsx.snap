// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="normal"
>
  <div
    className="welcome"
  />
  <ul
    className="list"
  >
    <li>
      To get started, edit 
      <code>
        src/pages/index.js
      </code>
       and save to reload.
    </li>
    <li>
      <div
        id="intro"
      >
        Welcome to Umi
      </div>
      <button
        onClick={[Function]}
      >
        Change
      </button>
    </li>
  </ul>
</div>
`;

exports[`Page: index matches snapshot 1`] = `
<div
  className="normal"
>
  <div
    className="welcome"
  />
  <ul
    className="list"
  >
    <li>
      To get started, edit 
      <code>
        src/pages/index.js
      </code>
       and save to reload.
    </li>
    <li>
      <div
        id="intro"
      >
        Welcome to Umi
      </div>
      <button
        onClick={[Function]}
      >
        Change
      </button>
    </li>
  </ul>
</div>
`;
